### PR TITLE
DSD-199: RadioGroup component - part 2

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-199](https://jira.nypl.org/browse/DSD-199)

## This PR does the following:
- Continues off from PR #661 
- This covers some concerns from the first PR and continues working through the Chakra refactor.
- Instead of using Chakra's own `RadioGroup` component, this uses other Chakra components and the style object to divide and appropriately set styles.
- Configures the `RadioGroup` component so it works with the DS `Radio` component.

## How has this been tested?

Locally in Storybook and linked in the Library Card app for uncontrolled and controlled forms states.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 I think the `CheckboxGroup` component will follow a very similar pattern for the DS version.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr664-0kvlxsozcctnvucajhh11k9tdbasmv6o.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/612ff43208cf47be295af8ae

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr664-0kvlxsozcctnvucajhh11k9tdbasmv6o.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/612ff43208cf47be295af8ae

File: src/components/RadioGroup/RadioGroup.tsx [hunk 1]
```diff
@@ -1,114 +1,148 @@
...
-        attributes["disabled"] = true;
-        attributes["aria-disabled"] = true;
-      }
-      if (errored) {
-        attributes["errored"] = true;
-      }
-      if (child.props.value === defaultValue) {
-        attributes["checked"] = true;
+    // Go through the Radio children and update them as needed.
+    React.Children.map(children, (child: React.ReactElement, i) => {
```
Comment (bigfishdesign13) [inline src/components/RadioGroup/RadioGroup.tsx:88]: Can you add validation here to ensure that only DS `Radio` components are being passed?

File: src/components/RadioGroup/RadioGroup.tsx [hunk 2]
```diff
@@ -1,114 +1,148 @@
...
+          <span>{labelText}</span>
+          {optReqFlag && (
+            <Box as="span" __css={styles.required}>
+              {isRequired ? "Required" : "Optional"}
+            </Box>
+          )}
+        </legend>
+        <Stack
+          direction={[direction]}
+          spacing={spacing.s}
```
Comment (bigfishdesign13) [inline src/components/RadioGroup/RadioGroup.tsx]: When `direction` is set to `row`, the spacing should be `2rem`, which aligns with `spacing.l`.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
This looks really good and I am excited for it.  Just needs a few small changes.

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr664-0kvlxsozcctnvucajhh11k9tdbasmv6o.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/612ff43208cf47be295af8ae

**Review by bigfishdesign13 — COMMENTED** [review]
I think this looks good, but I will allow @nypl-wluisi or @hokei to give the final approval on this one.

**Review by nypl-wluisi — APPROVED** [review]
This looks good to me.

**Review by bigfishdesign13 — APPROVED** [review]
My official approval on the PR.

**Pull request merged**